### PR TITLE
Implement Category Click Code

### PR DESCRIPTION
**Pull request by Tonnie-Dev**
I have added a `dao query` to filter out cached data and return meals by the selected category. 

Have also added a use case for this task which lead to many arguments on the ViewModel. For this, I did a little clean-up and moved the use cases to a single container.

My concern is the discrepancy when it comes to clicking a CategoryItem  - clicking on the upper end of the CategoryItem doesn't trigger the click but Clicking on the lower part triggers the click.

### CODE REVIEW

File: app/src/main/java/com/flexcode/yummy/domain/use_case/UseCaseContainer.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.flexcode.yummy.domain.use_case
```
Comment (Felix-Kariuki) [inline app/src/main/java/com/flexcode/yummy/domain/use_case/UseCaseContainer.kt:1]: Great Wrapping the Usecases together

File: app/src/main/java/com/flexcode/yummy/presentation/meals_screen/MealsViewModel.kt [hunk 2]
```diff
@@ -17,15 +13,15 @@ import kotlinx.coroutines.Job
...
 import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
 
 @HiltViewModel
 class MealsViewModel @Inject constructor(
-    private val getMealsUseCase: GetMealsUseCase,
-    private val getCategoriesUseCase: GetCategoriesUseCase
+    private val container: UseCaseContainer
```
Comment (Felix-Kariuki) [inline app/src/main/java/com/flexcode/yummy/presentation/meals_screen/MealsViewModel.kt:22]: this is great you can rename the container to usecaseContainer for easier understandability and readability

**Review by Felix-Kariuki — COMMENTED** [review]
Great Work. Rename the container to usecaseContainer as indidated.👏👏 Keep them PR's coming'🥳

File: app/src/main/java/com/flexcode/yummy/presentation/meals_screen/MealsViewModel.kt (hunk 2 above)
Comment (Tonnie-Dev) [reply app/src/main/java/com/flexcode/yummy/presentation/meals_screen/MealsViewModel.kt]: I have refactored the parameter to `useCaseContainer `and have pushed the code. Let me know if that is satisfactory.

**Pull request merged**